scipy/optimize/_direct.py

```diff
@@ -0,0 +1,203 @@
+        Prints the evaluated `func` at every iteration.
+    iatol : float, optional
+        Ensures sufficient decrease in function value when a new potentially
+        optimal hyperrectangle is chosen. Default is 0.0001.
+    maxfun : int, optional
+        Approximate upper bound on objective function evaluations.
+        Default is 2000.
+    maxiter : int, optional
+        Maximum number of iterations. Default is 6000.
+    locally_biased : bool, optional
+        Whether to use the original [1]_ or modified [2]_ DIRECT algorithm.
+        Default is True. Possible values:
+
+        * ``locally_biased=False`` - use the original DIRECT algorithm
+        * ``locally_biased=True`` - use the modified DIRECT-l algorithm
+    f_min : float, optional
+        Function value of the global optimum.
+        By default it is a negative value whose absolute value is very large.
+        Set this value only if the global optimum is known.
+    f_min_per : float, optional
+        Terminate the optimization when the percent error satisfies:
+
+        .. math::
+
+            100*(f_{min} - f_{global})/\max(1, |f_{global}|) \leq f_{glper}
+
+        Default is 0.01.
+    vol_per : float, optional
+        Terminate the optimization once the volume of a hyperrectangle is less
+        than vol_per percent of the original hyper-rectangle.
+        Default is -1.
+    sigma_per : float, optional
+        Terminate the optimization once the measure of the hyperrectangle is
+        less than this argument. Default is -1.
+    callback : callable, `callback(xk)`, optional
+        A function to follow the progress of the minimization. ``xk`` is
+        the best solution found so far.
+
+    Returns
+    -------
+    res : OptimizeResult
+        The optimization result represented as a ``OptimizeResult`` object.
+        Important attributes are: ``x`` the solution array, ``success`` a
+        Boolean flag indicating if the optimizer exited successfully and
+        ``message`` which describes the cause of the termination. See
+        `OptimizeResult` for a description of other attributes.
+
+    Notes
+    -----
+
+    DIRECT is a deterministic
+    optimization algorithm capable of minimizing a black box function with
+    its variables subject to lower and upper bound constrains by sampling
+    potential solutions in the search space. The algorithm starts by
+    normalising the hyperrectangle (set of all possible values that can
+    be taken by the input variables subject to the bound constraints) to
+    an n-dimensional unit hypercube. It samples the function at the
+    center of this hypercube and at 2n (n is the number of variables)
+    more points, 2 in each coordinate direction. Using these function
+    values, DIRECT then divides the domain into hyperrectangles, each
+    having exactly one of the sampling points as its center. In each
+    iteration, DIRECT chooses, using the epsilon parameter, which defaults
```

Comment:
Has epsilon been renamed? There is not a parameter named epsilon.
If the parameter is not exposed, then "default" would be misleading, as it suggests that it can be changed.